src/settings/mod.rs

```diff
@@ -1,29 +1,48 @@
 //! Booster NGFW NVM settings
 
-pub mod channel_settings;
+use core::fmt::Write;
+use heapless::String;
+use miniconf::Tree;
+
+pub mod eeprom;
 pub mod flash;
-pub mod global_settings;
 pub mod runtime_settings;
-mod sinara;
-use encdec::{Decode, DecodeOwned, Encode};
-use serde::{Deserialize, Serialize};
 
-use sinara::{BoardId as SinaraBoardId, SinaraConfiguration};
+use eeprom::main_board::{Cidr, IpAddr};
+use runtime_settings::RuntimeSettings;
+
+#[derive(Clone, Debug, Tree)]
+pub struct Settings {
+    #[tree(depth = 4)]
+    pub booster: RuntimeSettings,
 
-pub use channel_settings::BoosterChannelSettings;
-pub use global_settings::BoosterSettings;
+    #[tree(skip)]
+    pub mac: smoltcp_nal::smoltcp::wire::EthernetAddress,
 
-/// A semantic version control for recording software versions.
-#[derive(Encode, DecodeOwned, Serialize, Deserialize, Debug, PartialEq, Copy, Clone)]
-pub struct SemVersion {
-    major: u8,
-    minor: u8,
-    patch: u8,
+    pub ip: Cidr,
+    pub broker: String<255>,
+    pub gateway: IpAddr,
+    pub id: String<23>,
 }
 
-impl SemVersion {
-    /// Determine if this version is compatible with `rhs`.
-    pub fn is_compatible_with(&self, rhs: &SemVersion) -> bool {
-        (self.major == rhs.major) && (self.minor <= rhs.minor)
+impl serial_settings::Settings<5> for Settings {
+    fn reset(&mut self) {
+        self.id.clear();
+        write!(
```

Comment:
We should probably just use `Display`.

https://docs.rs/smoltcp/latest/src/smoltcp/wire/ethernet.rs.html#71

Reply:
Huh. I'm not sure why I did it any other way.

Reply:
Fixed in bb31b94a405c36a14f27e18bd903909ff96a41de